src/test/enginesynctest.cpp

```diff
@@ -1768,3 +1768,33 @@ TEST_F(EngineSyncTest, ChangeBeatGrid) {
     EXPECT_FLOAT_EQ(65.0, ControlObject::get(ConfigKey(m_sGroup2, "bpm")));
     EXPECT_FLOAT_EQ(130.0, ControlObject::get(ConfigKey(m_sInternalClockGroup, "bpm")));
 }
+
+TEST_F(EngineSyncTest, BeatMapQantizePlay) {
+    // This test demonstates https://bugs.launchpad.net/mixxx/+bug/1874918
+    BeatsPointer pBeats1 = BeatFactory::makeBeatGrid(*m_pTrack1, 120, 0.0);
+    m_pTrack1->setBeats(pBeats1);
+
+    BeatsPointer pBeats2 = BeatsPointer(new BeatMap(*m_pTrack2, 44100));
+    // Add two beats at 120 Bpm
+    pBeats2->addBeat(44100 / 2);
+    pBeats2->addBeat(44100);
+    m_pTrack2->setBeats(pBeats2);
+
+    ControlObject::set(ConfigKey(m_sGroup1, "quantize"), 1.0);
+    ControlObject::set(ConfigKey(m_sGroup2, "quantize"), 1.0);
+
+    // Make Channel1 master to weed out any channel ordering issues.
```

Comment:
remove this comment, it doesn't apply because we now always process the master enginebuffer first